### PR TITLE
Update SLMN.GG to use Vue 3 with all the consequences that come from that

### PR DESCRIPTION
**Pull request by zusorio**

### CODE REVIEW

**Comment by slmnio** [comment]
Vue 3 features

- [ ] Fully refactor utils/dashboard.js to remove all passed auth objects
- [ ] Logged in user dropdown is still being annoying
- [ ] Check all Bootstrap components are working with the new package
- [ ] Move fully to vue 3 off the compat build

2.0 features

- [ ] Data server should needs to full load tables from time to time
- [ ] SLMN.GG Learn needs to be fully complete

**Comment by slmnio** [comment]
Bunch of rules disabled because they were annoying but is probably worth enabling at some point for code quality

- typescript-eslint/no-floating-promises
- typescript-eslint/no-misused-promises
- typescript-eslint/no-implied-eval

File: website/src/components/broadcast/Squeezable.vue [hunk 1]
```diff
@@ -32,6 +32,8 @@ export default {
     methods: {
         updateWidth(isAfterTick) {
             const big = this.$refs.big;
+            // TODO: fix this - vue3 thing?
+            // eslint-disable-next-line vue/require-slots-as-functions
             const small = this.$slots.default?.[0]?.elm;
```
Comment (slmnio) [inline website/src/components/broadcast/Squeezable.vue:37]: From rule `vue/require-slots-as-functions` - need to look this up and fix it

File: website/src/components/broadcast/TeamAudio.vue [hunk 2]
```diff
@@ -97,9 +97,13 @@ export default {
...
                     console.log("garbage collecting", user);
                     this.players[user].destroy();
+                    // TODO: fix these
+                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
                     delete this.players[user];
                     this.decoders[user].free();
+                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
                     delete this.decoders[user];
+                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
                     delete this.lastPacketTime[user];
```
Comment (slmnio) [inline website/src/components/broadcast/TeamAudio.vue:107]: From rule `typescript-eslint/no-dynamic-delete`
This is from your audio stuff @zusorio, see if there's nice ways of unsetting without using delete

File: website/src/views/sub-views/PlayerMain.vue [hunk 3]
```diff
@@ -126,6 +126,8 @@ export default {
 
             Object.entries(groups).forEach(([key, val]) => {
                 if (val.items.length === 0) {
+                    // TODO: no-dynamic-delete fix
+                    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
                     delete groups[key];
```
Comment (slmnio) [inline website/src/views/sub-views/PlayerMain.vue:131]: From rule `typescript-eslint/no-dynamic-delete`
I think this is easy, just requires filtering further later

**Review by slmnio — CHANGES_REQUESTED** [review]
Couple bits I found that got stuck in the linter

File: website/src/components/broadcast/break/BreakOverlay.vue [hunk 4]
```diff
@@ -330,260 +350,313 @@ export default {
...
-    .break-transition-top {
-        height: 160px !important;
-    }
-
-    .break-transition-main {
-        height: 0 !important;
-        flex-grow: 1;
-        margin-top: 80px !important;
-    }
+.event-theme-border {
```
Comment (slmnio) [inline website/src/components/broadcast/break/BreakOverlay.vue:353]: My editor is definitely going to indent this again lol

Comment (zusorio) [reply website/src/components/broadcast/break/BreakOverlay.vue:353]: I should probably figure out why my editor is not doing it

Comment (slmnio) [reply website/src/components/broadcast/break/BreakOverlay.vue:353]: I had to force it, similar with tbody/thead indenting

**Review by slmnio — APPROVED** [review]
Holy send

**Review by slmnio — APPROVED** [review]

**Comment by zusorio** [comment]
Now tracked in #226

**Pull request closed without merging**